packages/fast-check/src/check/precondition/PreconditionFailure.ts

```diff
@@ -4,7 +4,9 @@
  * @public
  */
 export class PreconditionFailure extends Error {
+  /** @internal */
   private static readonly SharedFootPrint: symbol = Symbol('fast-check/PreconditionFailure');
```

Comment:
Probably worth changing it into a Symbol.for so that we make sure that CJS and MJS world can interoperate between each others.

Reply:
private static is not on the exported interface

Reply:
sadly symbol.for is also no fix;
```ts
const a = Symbol.for("a")
const b = Symbol.for("a")
const c = (_: typeof b) => 1

c(a)
//^ Argument of type 'typeof a' is not assignable to parameter of type 'typeof b'.
```

Reply:
(sorry, my previous message about not being on the interface was incorrect. it's not on there because of @internal :D

Reply:
yea separate would be great.
the current implementation is tested and works in the @effect/schema case